test/scripts/e2e_subs/e2e-app-simulate.sh

```diff
@@ -288,3 +288,76 @@ if [[ $(echo "$RES" | jq '."txn-groups"[0]."txn-results"[0]."txn-result"."logs"[
     date '+app-simulate-test FAIL the app call to logs-a-lot.teal for unlimited_log_test()void should succeed %Y%m%d_%H%M%S'
     false
 fi
+
+############################################################
+# WE WANT TO FURTHER TEST EXTRA BUDGET IN SIMULATION WORKS #
+############################################################
+
+function generate_teal() {
+    FILE=$1
+    VERSION=$2
+    REPETITION=$3
+
+    printf '#pragma version %d\n txn ApplicationID\n bz end\n' $VERSION > "${FILE}"
+
+    # iterating in interval [0, REPETITION - 1]
+    for i in $(seq 0 1 $(expr $REPETITION - 1)); do
+        printf "int 1\npop\n" >> "${FILE}"
+    done
+
+    printf "end:\n int 1\n" >> "${FILE}"
+}
+
+BIG_TEAL_FILE="$TEMPDIR/int-pop-400-cost-a-lot.teal"
+generate_teal "$BIG_TEAL_FILE" 8 400
+
+printf '#pragma version 8\nint 1' > "${TEMPDIR}/simple-v8.teal"
+
+RES=$(${gcmd} app create --creator ${ACCOUNT} --approval-prog "${BIG_TEAL_FILE}" --clear-prog "${TEMPDIR}/simple-v8.teal" --extra-pages 1 2>&1 || true)
+EXPSUCCESS='Created app with app index'
+if [[ $RES != *"${EXPSUCCESS}"* ]]; then
+    date '+app-simulate-test FAIL the app creation for generated large TEAL should succeed %Y%m%d_%H%M%S'
+    false
+fi
+
+APPID=$(echo "$RES" | grep Created | awk '{ print $6 }')
+
+# SIMULATION! without extra budget should fail direct call
+${gcmd} app call --app-id $APPID --from $ACCOUNT 2>&1 -o "${TEMPDIR}/no-extra-opcode-budget.tx"
+${gcmd} clerk sign -i "${TEMPDIR}/no-extra-opcode-budget.tx" -o "${TEMPDIR}/no-extra-opcode-budget.stx"
+RES=$(${gcmd} clerk simulate -t "${TEMPDIR}/no-extra-opcode-budget.stx")
+
+if [[ $(echo "$RES" | jq '."txn-groups" | any(has("failure-message"))') != $CONST_TRUE ]]; then
+    date '+app-simulate-test FAIL the app call to generated large TEAL without extra budget should fail %Y%m%d_%H%M%S'
+    false
+fi
+
+EXPECTED_FAILURE='dynamic cost budget exceeded'
+
+if [[ $(echo "$RES" | jq '."txn-groups"[0]."failure-message"') != *"${EXPECTED_FAILURE}"* ]]; then
+    date '+app-simulate-test FAIL the app call to generated large TEAL should fail %Y%m%d_%H%M%S'
+    false
+fi
+
+# SIMULATION! with extra budget should pass direct call
+RES=$(${gcmd} clerk simulate --extra-opcode-budget 200 -t "${TEMPDIR}/no-extra-opcode-budget.stx")
```

Comment:
Perhaps test with `allow-extra-opcode-budget` as well?

Reply:
see [more test in e2e for --allow-extra-opcode-budget, pr comment response](https://github.com/algorand/go-algorand/pull/5354/commits/2eebb810ca2052d22315fee59faa5f8aadd89940)